dev_guide/builds/advanced_build_operations.adoc

```diff
@@ -122,3 +122,132 @@ endif::[]
 If the specified `NodeSelector` cannot be matched to a node with those labels,
 the build still stay in the `Pending` state indefinitely.
 ====
+
+[[dev-guide-chaining-builds]]
+== Chaining Builds
+
+For compiled languages (Go, C, C++, Java, etc.), including the dependencies necessary for compilation in the application image
+might increase the size of the image or introduce vulnerabilities that can be exploited.
+
+To avoid these problems, two builds can be chained together. One that will produce the compiled artifact, and a second build
+that will place that artifact in a separate image that will run the artifact.
+
+In the following example, a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[Source-to-Image]
+build is combined with a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#docker-build[Docker]
+build to compile an artifact that is then placed in a separate runtime image.
```

Comment:
"Note: although this example chains an s2i build and a docker build, the first build can use any strategy that will produce an image containing the desired artifacts, and the second build can use any strategy that can consume input content from an image."

(wordsmith as desired).